hypershift-operator/controllers/hostedcluster/internal/platform/platform.go

```diff
@@ -49,19 +52,26 @@ type Platform interface {
 	ReconcileSecretEncryption(ctx context.Context, c client.Client, createOrUpdate upsert.CreateOrUpdateFN,
 		hcluster *hyperv1.HostedCluster,
 		controlPlaneNamespace string) error
+
+	// CAPIProviderPolicyRules responsible to return list of policy rules are required to be used
+	// by the CAPI provider in order to manage the resources by this platform
+	// Return nil if no aditional policy rule is required
+	CAPIProviderPolicyRules() []rbacv1.PolicyRule
 }
 
 func GetPlatform(hcluster *hyperv1.HostedCluster) (Platform, error) {
 	var platform Platform
 	switch hcluster.Spec.Platform.Type {
 	case hyperv1.AWSPlatform:
-		platform = aws.AWS{}
+		platform = &aws.AWS{}
```

Comment:
why this change?

Reply:
This is done for this comment: https://github.com/openshift/hypershift/pull/779#discussion_r777453439
The parameter passed in this function is interface, so changed the struct assignment to pointer

Reply:
oh, sorry I was referring to the reconcileCAPIProvider signature where we could pass p as pointer instead, just a minor thought. I can't see how does this change makes any difference?

Reply:
Yes, but p passed in the function is interface, you shouldn't change the interface to pointer but you can assign pointer to it, I think that do the job and prevent the copy, am I wrong?